### PR TITLE
Move session setup

### PR DESCRIPTION
**Pull request by antonrohr**
1st commit: Move session setup related classes into component SessionSetup.

My idea behind this was to clean up the somewhat crowded component OrbitQt by extracting files into a new SessionSetup component. Now that I did this change, I am questioning if this is a good idea. Please let me know what you think.

2nd commit: Rename ProfilingTargetDialog to SessionSetupDialog.
3rd commit: Rename files `servicedeploymanager.*` to `ServiceDeployManager.*`

### CODE REVIEW

File: src/Orbit/main.cpp [hunk 1]
```diff
@@ -193,9 +193,9 @@ static std::optional<std::string> GetCollectorPath(const QProcessEnvironment& pr
   return std::nullopt;
 }
 
-static orbit_qt::DeploymentConfiguration FigureOutDeploymentConfiguration() {
+static DeploymentConfiguration FigureOutDeploymentConfiguration() {
```
Comment (beckerhe) [inline src/Orbit/main.cpp:196]: Would it make sense to move this function into the new session setup module?

File: src/SessionSetup/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+          ClientServices
+          CONAN_PKG::grpc
+          CONAN_PKG::abseil
+          MetricsUploader
+          OrbitBase
+          OrbitGgp
+          OrbitSsh
+          OrbitSshQt
+          Qt5::Widgets
+          QtUtils)
```
Comment (beckerhe) [inline src/SessionSetup/CMakeLists.txt:57]: Could you put our internal targets at the top and sort them alphabetically?

**Review by beckerhe — APPROVED** [review]
I think it makes a lot of sense. LGTM!

File: src/Orbit/main.cpp (hunk 1 above)
Comment (antonrohr) [reply src/Orbit/main.cpp]: yes, done!

File: src/SessionSetup/CMakeLists.txt (hunk 2 above)
Comment (antonrohr) [reply src/SessionSetup/CMakeLists.txt]: done

Comment (beckerhe) [reply src/SessionSetup/CMakeLists.txt]: thx

**Pull request merged**